Sources/XcodeGenKit/ProjectGenerator.swift

```diff
@@ -49,13 +49,14 @@ public class ProjectGenerator {
             return XCScheme.BuildAction.Entry(buildableReference: buildableReference, buildFor: buildTarget.buildTypes)
         }
 
-        let testBuildTargets = tests.map {
+        let testableNames = scheme.test?.testables ?? []
+        let testablesBuildTargets = testableNames.map {
             Scheme.BuildTarget(target: $0, buildTypes: BuildType.testOnly)
         }
 
-        let testBuildTargetEntries = testBuildTargets.map(getBuildEntry)
+        let testBuildTargetEntries = testablesBuildTargets.map(getBuildEntry)
 
-        let buildActionEntries: [XCScheme.BuildAction.Entry] = scheme.build.targets.map(getBuildEntry) + testBuildTargetEntries
```

Comment:
Does this fix the issue you mention in the description?

Reply:
Yep, it does.

Reply:
Ok. In this case logic needs to be added so that if a TargetScheme defines some tests, they also get added as buildActionEntries.
Down where the Scheme is initialised, the tests need to be appended to the target.

Reply:
they actually are (better seen in split mode):
https://github.com/yonaskolb/XcodeGen/pull/195/files#diff-bcdced0f751833616e896b5b1e6e0fa8R118
and 
https://github.com/yonaskolb/XcodeGen/pull/195/files#diff-bcdced0f751833616e896b5b1e6e0fa8R129

Reply:
From what I can see it's passing the test targets in, but they are no longer automatically added as build entries, so you also need to pass them in as targets

Reply:
If you're talking about build entries as shown in "Scheme/Build/Targets" - Xcode automatically adds anything registered as testable there, quote from the PR description:
>  if we have 2 targets `STCore` and `STCoreTests`, if `STCoreTests` is also added as testable Xcode duplicates test target in build actions (i.e. `STCore`, `STCoreTests`,
`STCoreTests` are shown).

Reply:
Ah I see. By automatically add, does it create an extra diff in the project, or just display them in the UI?

Reply:
No diff, it just displays them in the UI